pupil-spa/src/app/services/check-complete/check-complete.service.spec.ts

```diff
@@ -67,24 +75,34 @@ describe('CheckCompleteService', () => {
     });
     it('submit should call submission service unsuccessfully and audit failure', async () => {
       const addEntrySpy = spyOn(auditService, 'addEntry');
+      spyOn(storageService, 'getItem').and.callFake(arg => getItemMock(arg, false));
       spyOn(submissionService, 'submitData')
-        .and.returnValue({ toPromise: () => Promise.reject(new Error('error')) });
+        .and.returnValue({toPromise: () => Promise.reject(new Error('error'))});
       await checkCompleteService.submit(Date.now());
       expect(addEntrySpy).toHaveBeenCalledTimes(1);
       expect(addEntrySpy.calls.all()[0].args[0].type).toEqual('CheckSubmissionApiCalled');
       expect(submissionService.submitData).toHaveBeenCalledTimes(1);
     });
+    it('submit should return if the app is configured to run in practice mode', async () => {
+      const addEntrySpy = spyOn(auditService, 'addEntry');
+      spyOn(storageService, 'getItem').and.callFake(arg => getItemMock(arg, true));
+      spyOn(submissionService, 'submitData');
+      await checkCompleteService.submit(Date.now());
+      expect(addEntrySpy).toHaveBeenCalledTimes(0);
+      expect(submissionService.submitData).toHaveBeenCalledTimes(0);
```

Comment:
I'm just trying to decide whether `.toHaveBeenCalledTimes(0);` would look better as `.not.toHaveBeenCalled()` - but that's a pretty minor comment really.